src/index.js

```diff
@@ -12,23 +12,49 @@ const propTypes = {
   level: PropTypes.oneOf(["L", "M", "Q", "H"]),
   size: PropTypes.number,
   value: PropTypes.string.isRequired,
+  withDownload: Proptypes.bool
 };
 
 const defaultProps = {
   bgColor: "#FFFFFF",
   fgColor: "#000000",
   level: "L",
   size: 256,
+  withDownload: false
 };
 
-const QRCode = ({ bgColor, fgColor, level, size, value }) => {
+const QRCode = ({ bgColor, fgColor, level, size, value, withDownload }) => {
   // We'll use type === -1 to force QRCode to automatically pick the best type.
   const qrcode = new QRCodeImpl(-1, ErrorCorrectLevel[level]);
   qrcode.addData(value);
   qrcode.make();
   const cells = qrcode.modules;
   const tileSize = size / cells.length;
+  
+  const downloadQR = () => {
```

Comment:
Is there any reason why all this is in this file? Once you have an `id` of the `svg` element, you can do all the logic outside of this lib. Right? How about moving all this code to a Storybook demo?